packages/api/src/dao/ValidatorsDAO.js

```diff
@@ -22,19 +37,59 @@ module.exports = class ValidatorsDAO {
     const fromRank = ((page - 1) * limit) + 1
     const toRank = fromRank + limit - 1
 
-    const subquery = this.knex('validators')
-      .select(this.knex('validators').count('pro_tx_hash').as('total_count'),
-        'validators.pro_tx_hash as pro_tx_hash', 'id')
-      .select(this.knex.raw(`rank() over (order by id ${order}) rank`))
+    const validatorsSubquery = this.knex('validators')
+      .select(
+        'validators.pro_tx_hash as pro_tx_hash',
+        'validators.id',
+        this.knex('validators').count('pro_tx_hash').as('total_count'),
+        this.knex('blocks')
+          .count('*')
+          .whereRaw('blocks.validator = validators.pro_tx_hash')
+          .as('proposed_blocks_amount'),
+        this.knex('blocks')
+          .select('hash')
+          .whereRaw('pro_tx_hash = blocks.validator')
+          .orderBy('height', 'desc')
+          .limit(1)
+          .as('proposed_block_hash')
+      )
       .as('validators')
 
+    const subquery = this.knex(validatorsSubquery)
+      .select(
+        'pro_tx_hash',
+        'id',
+        'total_count',
+        'proposed_blocks_amount',
+        this.knex.raw(`rank() over (order by validators.id ${order}) as rank`),
+        'blocks.hash as block_hash',
+        'blocks.height as latest_height',
+        'blocks.timestamp as latest_timestamp',
+        'blocks.l1_locked_height as l1_locked_height',
+        'blocks.app_version as app_version',
+        'blocks.block_version as block_version'
+      )
+      .leftJoin('blocks', 'blocks.hash', 'proposed_block_hash')
```

Comment:
Why different joins used in getValidatorByProTxHash and getValidators?

Reply:
✅